NOISSUE - Return certs on bootstrap view response

SammyOina opened the pull request:
### What does this do?
- Retrieves Client_Cert, and CA_Cert when a view request is made on the bootstrap service.
- Returns certs on updateCert
- Use named parameters on queries

### Which issue(s) does this PR fix/relate to?
No issue

### List any changes that modify/break current functionality

### Have you included tests for your changes?

### Did you document any new/modified functionality?
I have added modified parameters to openapi spec

### Notes

SammyOina committed:
return certs on bootstrap view response

Signed-off-by: SammyOina <sammyoina@gmail.com>

dborovcanin commented on bootstrap/postgres/configs.go:
```diff
@@ -86,7 +86,7 @@ func (cr configRepository) Save(ctx context.Context, cfg bootstrap.Config, chsCo
 
 func (cr configRepository) RetrieveByID(ctx context.Context, owner, id string) (bootstrap.Config, error) {
-	q := `SELECT mainflux_thing, mainflux_key, external_id, external_key, name, content, state
+	q := `SELECT mainflux_thing, mainflux_key, external_id, external_key, name, content, state, client_cert, client_key, ca_cert
```
We should probably not return the client key.

SammyOina replied:
ok, I'll leave to be returned on updateCert only

SammyOina committed:
return updated certs when updated

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
fix test

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
fix test

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
fix test

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
fix test

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
fix tests

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
simplify tests

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
use named query

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
fix test

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
use named params

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
fix typo

Signed-off-by: SammyOina <sammyoina@gmail.com>

dborovcanin commented on provision/service.go:
```diff
@@ -226,7 +226,8 @@ func (ps *provisionService) Provision(token, name, externalID, externalKey strin
 			if needsBootstrap(thing) {
-				if err = ps.sdk.UpdateBootstrapCerts(bsConfig.MFThing, cert.ClientCert, cert.ClientKey, "", token); err != nil {
+				_, err = ps.sdk.UpdateBootstrapCerts(bsConfig.MFThing, cert.ClientCert, cert.ClientKey, "", token)
+				if err != nil {
```
Inline.

dborovcanin commented on bootstrap/service_test.go:
```diff
@@ -288,32 +289,50 @@ func TestUpdateCert(t *testing.T) {
-		err := svc.UpdateCert(context.Background(), tc.token, tc.thingKey, tc.clientCert, tc.clientKey, tc.caCert)
+		cfg, err := svc.UpdateCert(context.Background(), tc.token, tc.thingKey, tc.clientCert, tc.clientKey, tc.caCert)
 		assert.True(t, errors.Contains(err, tc.err), fmt.Sprintf("%s: expected %s got %s\n", tc.desc, tc.err, err))
+		if err == nil {
```
Is this check needed? In case of an error, the config will be empty.

SammyOina committed:
use inline error checks
remove unrequired conditions

Signed-off-by: SammyOina <sammyoina@gmail.com>

drasko reviewed (dismissed):
LGTM

SammyOina committed:
sort slices before comparison

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
rename mainflux_id to thing_id
rename MFThing to ThingID
rename MFKey to ThingKey
rename mainflux_key to thing_key

Signed-off-by: SammyOina <sammyoina@gmail.com>

dborovcanin reviewed (changes requested):
Please fix CI remarks.

arvindh123 commented on bootstrap/configs.go:
```diff
@@ -16,13 +16,13 @@ import (
 	CACert      string    `json:"ca_cert,omitempty"`
-	MFKey       string    `json:"mainflux_key"`
+	ThingKey    string    `json:"thing_key"`
 	MFChannels  []Channel `json:"mainflux_channels,omitempty"`
```
@SammyOina, Is there any reason behind for not changing `mainflux_channels` to `channels` ?

SammyOina replied:
yes, the create request channels is of type []string and for view request it is of type []Channel

arvindh123 commented on bootstrap/api/responses.go:
```diff
@@ -67,14 +67,16 @@ type channelRes struct {
+	ThingID     string          `json:"thing_id,omitempty"`
+	ThingKey    string          `json:"thing_key,omitempty"`
 	Channels    []channelRes    `json:"mainflux_channels,omitempty"`
 	ExternalID  string          `json:"external_id"`
```
if `mainflux_id` can be changed to `thing_id`,  `mainflux_key` to `thing_key` , then `mainflux_channels` can be changed to `channels`

SammyOina replied:
@arvindh123 theres a clash between requests and responses, what do you suggest https://github.com/SammyOina/mainflux/blob/70f53c2f979dbe408d9b1ad7ba0ba99a72eb0c70/pkg/sdk/go/bootstrap.go#L30-L35

SammyOina committed:
remove mainflux_channels

Signed-off-by: SammyOina <sammyoina@gmail.com>

SammyOina committed:
simplify unmarshaller

Signed-off-by: SammyOina <sammyoina@gmail.com>

drasko reviewed (approved):
LGTM
